src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -1,952 +1,382 @@
-                .Include(dataBlockVersion => dataBlockVersion.DataBlockParent)
-                .Where(dataBlockVersion => dataBlockVersion.ReleaseVersionId == releaseVersionId)
-                .ToListAsync(cancellationToken);
-
-            var dataBlockParents = dataBlockVersions
-                .Select(dataBlockVersion => dataBlockVersion.DataBlockParent)
-                .Distinct()
-                .ToList();
-
-            // Unset the DataBlockVersion references from the DataBlockParent.
-            dataBlockParents.ForEach(dataBlockParent =>
-            {
-                dataBlockParent.LatestDraftVersionId = null;
-                dataBlockParent.LatestPublishedVersionId = null;
-            });
-
-            await _context.SaveChangesAsync(cancellationToken);
+            // Copy certain fields from the original.
+            Caption = originalSection.Caption,
+            Heading = originalSection.Heading,
+            Order = originalSection.Order,
+            Type = originalSection.Type
+        };
+    }
 
-            // Then remove the now-unreferenced DataBlockVersions.
-            _context.DataBlockVersions.RemoveRange(dataBlockVersions);
-            await _context.SaveChangesAsync(cancellationToken);
+    private async Task<Either<ActionResult, (Release release, string oldReleaseSlug, string newReleaseSlug)>> ValidateUpdateRequest(
+        Release release, 
+        ReleaseUpdateRequest releaseUpdate,
+        CancellationToken cancellationToken)
+    {
+        var newReleaseSlug = CalculateNewReleaseSlug(release, releaseUpdate);
+        var oldReleaseSlug = release.Slug;
+
+        return await ValidateReleaseSlugUniqueToPublication(
+            slug: newReleaseSlug,
+            publicationId: release.PublicationId,
+            releaseId: release.Id,
+            cancellationToken: cancellationToken)
+            .OnSuccess(async _ => await ValidateReleaseIsNotUndergoingPublishing(release.Id, cancellationToken ))
+            .OnSuccess(async _ => await ValidateReleaseRedirectDoesNotExistForNewSlug(
+                releaseId: release.Id,
+                newReleaseSlug: newReleaseSlug, 
+                cancellationToken: cancellationToken))
+            .OnSuccess(_ => (release, oldReleaseSlug, newReleaseSlug));
+    }
 
-            // And finally, delete the DataBlockParents if they are now orphaned.
-            var orphanedDataBlockParents = dataBlockParents
-                .Where(dataBlockParent =>
-                    !_context
-                        .DataBlockVersions
-                        .Any(dataBlockVersion => dataBlockVersion.DataBlockParentId == dataBlockParent.Id))
-                .ToList();
+    private static string CalculateNewReleaseSlug(Release release, ReleaseUpdateRequest releaseUpdate)
+    {
+        return NamingUtils.CreateReleaseSlug(
+            year: release.Year,
+            timePeriodCoverage: release.TimePeriodCoverage,
+            label: releaseUpdate.Label);
+    }
```

Comment:
I think this method is a bit unnecessary and that you could just inline `var newReleaseSlug = NamingUtils.CreateReleaseSlug...`.